src/analyzer/protocol/ssl/tls-handshake-analyzer.pac

```diff
@@ -272,15 +272,48 @@ refine connection Handshake_Conn += {
 		return true;
 		%}
 
-	function proc_ec_server_key_exchange(rec: HandshakeRecord, curve_type: uint8, curve: uint16) : bool
+	function proc_ecdhe_server_key_exchange(rec: HandshakeRecord, curve_type: uint8, curve: uint16, point: bytestring, signed_params: bytestring) : bool
 		%{
 		if ( curve_type == NAMED_CURVE )
 			BifEvent::generate_ssl_server_curve(bro_analyzer(),
 			  bro_analyzer()->Conn(), curve);
+			BifEvent::generate_ssl_ecdh_server_params(bro_analyzer(),
+			  bro_analyzer()->Conn(), curve, new StringVal(point.length(), (const char*)point.data()));
+			BifEvent::generate_ssl_server_signature(bro_analyzer(),
+			  bro_analyzer()->Conn(), new StringVal(signed_params.length(), (const char*)signed_params.data()));
 
 		return true;
 		%}
 
+	function proc_ecdh_anon_server_key_exchange(rec: HandshakeRecord, curve_type: uint8, curve: uint16, point: bytestring) : bool
+		%{
+		if ( curve_type == NAMED_CURVE )
+			BifEvent::generate_ssl_server_curve(bro_analyzer(),
+			  bro_analyzer()->Conn(), curve);
+			BifEvent::generate_ssl_ecdh_server_params(bro_analyzer(),
```

Comment:
Same comment as above applies here - this only works with named curves, right?

Reply:
Actually I just saw the comment in your other source file - I will assume that everything only works with named curves for now.